controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -494,10 +496,35 @@ func (r *BareMetalHostReconciler) actionDeleting(prov provisioner.Provisioner, i
 		return deleteComplete{}
 	}
 
+	// Only try to power off `maxPowerOffRetryCount` times before giving up, and deleting the host anyway
+	if info.host.Status.ErrorType != metal3v1alpha1.PowerManagementError || (info.host.Status.ErrorType == metal3v1alpha1.PowerManagementError && info.host.Status.ErrorCount < maxPowerOffRetryCount) {
+		info.log.Info("host ready to be powered off")
+		provResult, err := prov.PowerOff(
+			metal3v1alpha1.RebootModeHard,
+			info.host.Status.ErrorType == metal3v1alpha1.PowerManagementError)
+
+		if err != nil {
+			return actionError{errors.Wrap(err, "failed to power off")}
+		}
+
+		if provResult.ErrorMessage != "" {
+			return recordActionFailure(info, metal3v1alpha1.PowerManagementError, provResult.ErrorMessage)
+		}
+
+		if provResult.Dirty {
+			return actionContinue{provResult.RequeueAfter}
+		}
```

Comment:
If a PowerManagementError is set we should clear it here and reset host.Status.ErrorCount to 0.
We'll also need to set a flag to indicate that at the next return we want to return `actionUpdate`, not just `actionContinue` (otherwise the changes won't get written to the CR).